lib/facility.rb

```diff
@@ -1,14 +1,49 @@
+require "./lib/vehicle"
 class Facility
-  attr_reader :name, :address, :phone, :services
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
 
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  def initialize(facility_hash)
+    @name = facility_hash[:name]
+    @address = facility_hash[:address]
+    @phone = facility_hash[:phone]
     @services = []
+    @registered_vehicles = []
+    @collected_fees = 0
   end
 
-  def add_services(service)
+  def add_service(service)
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    @registered_vehicles << vehicle
+    @collected_fees += registration_fee(vehicle)
+    @registration_date = Date.today
+
+    return vehicle
+  end
+
+  def set_plate
+      if vehicle.antique?
+        vehicle.set_plate(:antique)
+      elsif vehicle.electric_vehicle?
+        vehicle.set_plate(:ev)
+      else
+        vehicle.set_plate(:regular)
+      end
+    end
```

Comment:
This helper method looks really great! I like that you made a helper method in the Vehicle class called `set_plate`